lib/rules/no-async-promise-executor.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @fileoverview disallow using an async function as a Promise executor
+ * @author Teddy Katz
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow using an async function as a Promise executor",
+            category: "Fill me in",
+            recommended: false,
+            url: "https://eslint.org/docs/rules/no-async-promise-executor"
+        },
+        fixable: null,
+        schema: []
+    },
+
+    create(context) {
+        return {
+            "NewExpression[callee.name='Promise'][arguments.0.async=true]"(node) {
+                context.report({
+                    node: context.getSourceCode().getFirstToken(node.arguments[0]),
```

Comment:
I imagine this wouldn't point at the right token if the node is surrounded by parens?

Reply:
I think it still would be. I'll add a test.

Reply:
Well, I'm wondering if we should just use `context.getSourceCode().getFirstToken(node.arguments[0], { filter: t => t.value === "async" })` or similar to be absolutely safe.